src/components/Conference.tsx

```diff
@@ -0,0 +1,211 @@
+} from '../services/ConferenceService';
+
+export interface IConferenceProps {
+    connect: () => ConferenceConnection;
+    room: string;
+}
+
+const userMediaConfig = {
+    audio: true,
+    video: true,
+}
+
+export class Conference extends React.Component<IConferenceProps, void> {
+    private connection: ConferenceConnection;
+    private peerConnections: { [id: string]: RTCPeerConnection };
+    private localStream: MediaStream;
+    private remoteStreams: { [id: string]: MediaStream };
+    private localId: string;
+
+    constructor() {
+        super();
+        this.connection = this.props.connect();
+        this.connection.subscribe(this.handleIncomingMessage);
+        this.handleIncomingMessage.bind(this);
+        this.joinRoom(this.props.room);
+        this.getUserMedia();
+    }
+
+    // TODO(yunsi): Complete display view.
+    public render() {
+        return (
+            <div />
+        )
+    }
+
+    public componentWillUnmount() {
+        this.leaveRoom()
+    }
+
+    private sendMessage(message: IConfOutgoingMessage) {
+        this.connection.publish(message);
+    }
+
+    private joinRoom(room: string) {
+        const message = createOutgoingMessageJoin(room);
+        this.sendMessage(message);
+    }
+
+    private leaveRoom() {
+        const message = createOutgoingMessageBye()
+        this.sendMessage(message);
+    }
+
+    private getUserMedia() {
+        navigator.mediaDevices.getUserMedia(userMediaConfig).then(stream => {
+            // TODO(yunsi): Send MediaStream to Local stream component.
+            this.localStream = stream;
+        })
+    }
+
+    private handleIncomingMessage(message: IConfIncomingMessage) {
+        switch (message.type) {
+            case 'Self':
+                return this.handleSelfMessage(message);
+            case 'AddPeer':
+                return this.handleAddPeerMessage(message);
+            case 'RemovePeer':
+                return this.handleRemovePeerMessage(message);
+            case 'Candidate':
+                return this.handleCandidateMessage(message);
+            case 'Offer':
+                return this.handleOfferMessage(message);
+            case 'Answer':
+                return this.handleAnswerMessage(message);
+            default:
+                return console.log('Unkonw message type')
+        }
+    }
+
+    private handleSelfMessage(message: IConfMessageSelf) {
+        this.localId = message.Id;
+    }
+
+    // NOTE(yunsi): When received an AddPeer event, conference will create a new PeerConnection and add it to the connection list.
+    private handleAddPeerMessage(message: IConfMessageAddPeer) {
+        const id = message.Id;
+        const peerConnection = this.createPeerConnectionById(id);
+
+        // NOTE(yunsi): When two clients both recieved an AddPeer event with the other client's id, they will do a compare to see who should create and send the offer.
+        if (this.localId.localeCompare(id)) {
```

Comment:
should be `this.localId.localeCompare(id) === 1` or `this.localId.localeCompare(id) === -1`